src/app/components/mentor-addition/mentor-addition.component.ts

```diff
@@ -0,0 +1,48 @@
+import {Component, Inject, OnInit} from '@angular/core';
+import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material/dialog';
+import {FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {MentorService} from '../../services/mentor.service';
+import {Mentor} from '../../shared/mentor';
+
+@Component({
+  selector: 'app-mentor-addition',
+  templateUrl: './mentor-addition.component.html',
+  styleUrls: ['./mentor-addition.component.scss']
+})
+export class MentorAdditionComponent implements OnInit {
+  mentorAdditionForm: FormGroup;
+  newMentor: Mentor = {name: '', stream: ''};
+  constructor(@Inject(MAT_DIALOG_DATA) public data: {str: string}, private fb: FormBuilder, private mentorService: MentorService,
+              public dialogRef: MatDialogRef<MentorAdditionComponent>) {
+    dialogRef.disableClose = true;
+  }
+
+  ngOnInit(): void {
+    this.mentorAdditionForm = this.fb.group({
+      name: ['', {validators: [Validators.required]}],
+      surname: ['', {validators: [Validators.required]}],
+    });
+  }
+  onSubmit() {
+    this.newMentor.name = this.mentorAdditionForm.value.name + ' ' + this.mentorAdditionForm.value.surname;
+    this.newMentor.stream = this.data.str;
+    this.mentorService.addMentor(this.newMentor).subscribe((mentor) => {this.newMentor.id = mentor.id ;
```

Comment:
It would be better to have this modal reusable and return the mentor as a result. Then the result could be sent to the service from the component, not the modal.
This is about what components are responsible for what. Usually a modal would not be responsible for saving the mentor, only collecting the data.